R/model.R

```diff
@@ -214,17 +215,41 @@ compile_method <- function(quiet = TRUE,
                            opencl = FALSE,
                            opencl_platform_id = 0,
                            opencl_device_id = 0,
-                           compiler_flags = NULL) {
-  exe <- strip_ext(self$stan_file())
+                           compiler_flags = NULL,
+                           force_recompile = FALSE) {
   make_local_changed <- set_make_local(threads,
                                        opencl,
                                        opencl_platform_id,
                                        opencl_device_id,
                                        compiler_flags)
+  exe <- cmdstan_ext(strip_ext(self$stan_file()))
+  recompile <- make_local_changed || force_recompile ||
+               !file.exists(exe) || (file.mtime(exe) < file.mtime(self$stan_file()))
```

Comment:
```
(file.mtime(exe) < file.mtime(self$stan_file()))
```

file.mtime(exe) is an NA if exe doesn't exist. So NAs work their way into the logic. I don't know how R treats NAs in logicals.

Maybe just break this out into more explicit ifs instead of ORing to avoid NAs. Unless of course if(NA) is defined and I'm just not familiar with R and this is all okay.

Reply:
So the check here is 
```R
!file.exists(exe) || (file.mtime(exe) < file.mtime(self$stan_file()))
```
If the exe file doesnt exist this is TRUE || NA which is TRUE. If it does exist, then the second clause will be either TRUE or FALSE and its fine.

```R
> TRUE || NA
[1] TRUE
> FALSE || NA
[1] NA

```

Reply:
I will make this more elaborate. All this "cleverness" hurts more then it gains.